File: Domain/Post.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Domain
{
    public class Post
    {
        public Guid Id { get; set; }
        public string Type { get; set; }
        public string Description { get; set; }
        public string City { get; set; }
        public string Region { get; set; }
        public string Location { get; set; }
        public DateTime Date { get; set; }
        public bool IsCancelled { get; set; }
        public ICollection<PostMember> Members { get; set; } // many-to-many relationship
        public string PosterEmail { get; set; }
        public ICollection<Photo> Photos { get; set; } // one-to-many relationship
    }
}

File: frontend/src/App/models/post.ts
import { Photo } from "./photo";
import { User } from "./user";

// Interface to define a model for post
export interface Post {
    id: string;
    type: string;
    description: string;
    city: string;
    region: string;
    location: string;
    date: string;
    posterName?: string;
    posterEmail?: string;
    members? : any[];
    photos?: any[];
    isCancelled?: boolean;
}

export interface Submit1 {
    type: string;
    city: string;
    region: string;
    location: string;
    date: string;
}

export interface JoinPost {
    id: string;
    email: string;
}